core/class/CommandLineTool.go

```diff
@@ -185,6 +190,41 @@ func (tool *CommandLineTool) placeInputFilesToCommandExecDir() error {
 	return nil
 }
 
+// defineStdoutDestination
+func (tool *CommandLineTool) defineStdinSource() error {
+	if tool.Root.Stdin != "" {
```

Comment:
should be early return. Deal special cases first and return earlier. Don't create huge `if` block. In this case,

```go
if tool.Root.Stdin == "" {
    return nil
}
// then, do something for `tool.Root.Stdin`
```